src/v/cluster/rm_stm.cc

```diff
@@ -1986,6 +1986,38 @@ void rm_stm::fill_snapshot_wo_seqs(T& snapshot) {
 }
 
 ss::future<stm_snapshot> rm_stm::take_snapshot() {
+    auto start_offset = _raft->start_offset();
+
+    std::vector<abort_index> abort_indexes;
+    std::vector<abort_index> expired_abort_indexes;
+    abort_indexes.reserve(_log_state.abort_indexes.size());
```

Comment:
How large can this vector be? Just curious if we're guarding against OOM, or need to add paging / chunking to split it up.

Reply:
yeh good question. it seems plausible that the size is indirectly related to client behavior, so worst case would be that it is susceptible to adversarial workloads?

Reply:
I think we can avoid this by storing `aborted_indexes` as a btree_map on `last_offset` -> entry and then doing a prefix truncation on std::upper_bound(start_offset)? (need to cleanup the snapshots before erasing)

Reply:
An assumption is that the users usually don't abort transactions so the abort index should be small. We store a vector of abort indexes as part of the snapshot o changing the type require changing an on disk format. Let's optimize it later if it's needed.

Reply:
@rystsov I'm fine leaving this as a vector, but

> We store a vector of abort indexes as part of the snapshot o changing the type require changing an on disk format.

this isn't really holding us back from changing the in-memory representation if it was truly a concern. it would be a bit of a hassle to fix, tho.